Feature/help-topic-drop-down (combotree)

kest874 opened the pull request:
Implements a combotree for the helptopic drop down:
# **Client Side:**

![image](https://cloud.githubusercontent.com/assets/12860950/16492861/ed15f452-3ea8-11e6-801d-ed23b2c14e01.png)

![image](https://cloud.githubusercontent.com/assets/12860950/16492865/f24c46d8-3ea8-11e6-9186-63d82da1dede.png)
# **Staff Side:**

**New Ticket:**

![image](https://cloud.githubusercontent.com/assets/12860950/16492959/6080c732-3ea9-11e6-8cc1-217ef22e0b75.png)

![image](https://cloud.githubusercontent.com/assets/12860950/16492965/65b8527e-3ea9-11e6-8cb3-8274d595f3f7.png)

**Updating a ticket:**

![image](https://cloud.githubusercontent.com/assets/12860950/16492979/74c56a72-3ea9-11e6-8b51-ab4a24459a22.png)

![image](https://cloud.githubusercontent.com/assets/12860950/16492988/7be78efc-3ea9-11e6-96c6-43de0f9fb138.png)

kest874 committed:
Merge pull request #76 from osTicket/develop

Merge upstream

kest874 committed:
Feature/Help Topic Dropdown (combotree)

Adds a combotree to the Help Topic Dropdown to ease use of selecting a Help Topic

kest874 commented:
Will need to cherry-pick  0c168d2 Not sure why the other commit is in there.

kest874 committed:
Opps: fix client side redactor and filedrop

kest874 committed:
Fixed issue where combo tree would not show the choosen helptopic when ?topicId= was specified

philbertphotos commented:
this is nice

rbukovansky commented:
@kest874 Very good job! This will be very useful for company I work for... **THUMBS UP**

arnascompensa commented:
It's really awesome, but I have one problem. I choose another language instead of English, it doesn't load anymore. Maybe someone has some suggestions?

kest874 commented:
Not sure what the issue might be, getting errors?  I changes my language to Spanish but it's still working fine.

kest874 committed:
Adds Help Topic Mass Change with the combotree.

Conflicts:
	include/ajax.tickets.php
	include/staff/templates/tickets-actions.tmpl.php

kest874 commented:
Added Help Topic Mass Change:

![image](https://cloud.githubusercontent.com/assets/12860950/16999455/a945dfa4-4e83-11e6-9684-1ac9ba6c8b2a.png)

![image](https://cloud.githubusercontent.com/assets/12860950/16999457/af9b015e-4e83-11e6-9282-c33bd2abe30c.png)

![image](https://cloud.githubusercontent.com/assets/12860950/16999460/b4995f20-4e83-11e6-819a-e76591e24cf9.png)

This will change all selected tickets to whatever help topic you choose.

kest874 committed:
Opps missing template file

ArthurBorsboom commented:
Hi,

Since there has been no commits or responses since July 2016 on the original osTicket project, I have decided to create a fork.

https://github.com/ArthurBorsboom/osTicket

There are many developers who have created awesome improvements by pull requests, which are eagerly waiting for merging.

This is an attempt to incorporate most pull requests to get create an improved version of osTicket. Hopefully this can later be reintegrated in the original project, or if the original project is dead as a rebranded project.

Would you mind to offer your pull request again to this fork?

Best regards,
Arthur Borsboom.

kest874 commented:
@ArthurBorsboom Ok with me.

ArthurBorsboom commented:
(same here)

In fact, Martin and I have chosen to create a non-personal fork, called osTicketFork.

https://github.com/osTicketFork/osTicket

I guess we will pick up your nice commit a bit later, or you offer it yourself to this fork.

pradeep1987oct22 commented:
Sir,
 Iam using OS Ticket with  version 1.10. I had followed the procedures of changing my help topic combo box for tree structure view.My problem is that his tree structure doesn't display (+ or -) images while selecting inside the help topic combo box.

The png files(combo_arrow.png,tree_icons.png) is been saved in /osticket/image 

Please help me


![screenshot from 2016-12-06 12-23-41](https://cloud.githubusercontent.com/assets/24405664/20916969/8e6a6e30-bbb0-11e6-92a9-6bef0cfc72d6.png)

Neangvuth commented:
Hi Guy,

I install OSticket V1.9.14. I want change combo list style in help topic, could you tell me which file that i need to edit?

Thanks

pradeep1987oct22 commented:
Dear  Neangvuth ,
 follow these steps   []( https://github.com/osTicket/osTicket/pull/3250/files)

pradeep1987oct22 commented:
https://github.com/osTicket/osTicket/pull/3250/files

Neangvuth commented:
Dear Guy,

I follow up these step already, but after that my OSticket not working.
I was changed code by these by. I changed 17 files.
Could you tell me how should i fix it?

![untitled](https://cloud.githubusercontent.com/assets/24447465/20998288/51649246-bd3f-11e6-9dab-d5520d31cf78.png)

ntozier commented:
@Neangvuth If you are getting a white screen then you should consult your PHP error log to see what the error is.  The mod Author and nor anyone else will be able to assist you with out knowing what the problem is.

pradeep1987oct22 commented:
Dear ntozier,Kindly support my issue.. iam struggling for it from past 2 days ,issue  has been raised (please go back  5 comments earlier)

ntozier commented:
@pradeep1987oct22 I cannot support your issue.  I am not the author of this mod.
If you want to use his mod you either have to:
A. install the version he wrote it for.
B. modify the mod so that it works for your version.

xdreamarun commented:
@ntozier 
As per steps we have integrated the code well.
Its working fine.
One issue we have faced.
Form - **Staff/Create New Ticket** -
while validating other fields like Issue summary - the help topic drop down selected values are resetting(Select Help Topic).
After clicking submit button without filling all required values, we are getting this issue.

xdreamarun commented:
@ntozier
we are using version 1.10.
we have integrated above code in this version.
while create a ticket Its validating Parent Help topic is selected or not. Can we validate If Child help Topic is selected or not. as per our business scenario user or agent must select Child help Topic.
Please help us..

ntozier commented:
@xdreamarun I am unsure why you are tagging me in your responses.  Please see my prior response in this thread.

xdreamarun commented:
@kest874
Dear SIr,
As per steps we have integrated the code well.
Its working fine.
One issue we have faced.
Form - Staff/Create New Ticket -
while validating other fields like Issue summary - the help topic drop down selected values are resetting(Select Help Topic).
After clicking submit button without filling all required values, we are getting this issue.

xdreamarun commented:
@kest874
Dear Sir,
,we are using version 1.10.
we have integrated above code in this version.
while create a ticket Its validating Parent Help topic is selected or not. Can we validate If Child help Topic is selected or not. as per our business scenario user or agent must select Child help Topic.
Please help us..

giannifusto commented:
@kest874
Dear Sir
we are using version 1.9.15 italian language.
 After editing the files for use Help topic drop down:
- css/helptopic.css 
- images/combo_arrow.png 
- images/tree_icons.png 
- include/ajax.tickets.php 
- include/class.forms.php 
- include/class.ticket.php 
- include/class.topic.php 
- include/client/header.inc.php 
- include/client/open.inc.php 
- include/staff/header.inc.php 
- include/staff/templates/tickets-actions.tmpl.php 
- include/staff/templates/topic.tmpl.php 
- include/staff/ticket-edit.inc.php 
- include/staff/ticket-open.inc.php 
- js/jquery.easyui.min.js 
- js/jquery.min.js 
- scp/css/scp.css 

I have this error 
Fatal error: Cannot redeclare class TicketModel in C:\inetpub\wwwroot\osticket\include\class.ticket.php on line 211 

How can I fix the problem?

kest874 commented:
> Fatal error: Cannot redeclare class TicketModel in C:\inetpub\wwwroot\osticket\include\class.ticket.php on line 211

Means you have 2 class TicketModel defined, delete one of them.

giannifusto commented:
Sorry but I'm a beginner with php. I attached file and please you can control? Thanks

[class.ticket.zip](https://github.com/osTicket/osTicket/files/729918/class.ticket.zip)

kest874 commented:
@giannifusto took a look and realized your not using 1.10 which this pull request is for.

khaza123 commented:
@pradeep1987oct22
I solved this problem. Get the appropriate image path to the helptopic.css file in combo_arrow.png and  tree_icons.png

thanks

NBorba commented:
@kest874 

Ken,

Thanks for your awesome work!

We've upgraded our system to 1.10 and implemented your code with no hassle. The only problem we are currently having is when a user creates a ticket and tries to attach a file to a ticket, it only brings them back to the top of the page and nothing else happens. That only occurs when trying to open a ticket on the client side, attaching files when opening and updating tickets on the staff side is normal. So, I've narrowed down two files (**header.inc.php** and **open.inc.php**) that when added to the system cause this behavior, however I haven't been able to find the specific cause, maybe some imports are missing, some code is missing. Need some help to fix this. 

Thanks!

kest874 commented:
I'll have to test but the lines you added to the header may need to be moved down I suspect that a later java script include is getting ignored.

NBorba commented:
@kest874 

Actually, the only thing I did was copy the 17 files needed for the dropdown feature to work. Didn't add nor remove from your code. I've noticed that only when those two files get copied over to the system that this behavior happens. I tried to check if some import was missing or if some code had been removed, but I haven't been able to find anything. Would appreciate your help! 

Thanks!

kest874 commented:
@NBorba can you try this, move in include/client/header.inc.php move line 48 and 49 below line 50 and see if it resolves the issue:
![image](https://cloud.githubusercontent.com/assets/12860950/22551367/29e70cc4-e91a-11e6-82fa-49df4af70f46.png)

NBorba commented:
Unfortunately that didn't work. I  also tried moving a bunch of scripts around inside the header and that didn't work either. 
I'm fairly certain it's something to do with the order of the scripts. Chrome's console gave me this error with both the original order and the order you told me to put the scripts in:

`Uncaught TypeError: Cannot read property 'regional' of undefined
    at HTMLDocument.<anonymous> (js:55)
    at j (jquery.min.js:2)
    at Object.fireWith [as resolveWith] (jquery.min.js:2)
    at Function.ready (jquery.min.js:2)
    at HTMLDocument.J (jquery.min.js:2)`

Hopefully that'll give you some hint.

pradeep1987oct22 commented:
Dear khaza123 

I had used correct path in helptopics.css

i had gone through the  code which contains ./images/blank.gif & ./images/panel_tools.png

Both the PNG  above are not present in original downloaded list of files (screenshot attached)
![screenshot from 2017-02-08 17-49-03](https://cloud.githubusercontent.com/assets/24405664/22737875/ed295b40-ee26-11e6-8f10-5adeb62c8398.png)

Kindly send the images files or the edited helptopic.css

kavehsafavi commented:
@kest874 
@NBorba
i have the same problem with attaching in client pages. The problem we are currently having is when a user creates a ticket and tries to attach a file to a ticket, it only brings them back to the top of the page and nothing else happens. That only occurs when trying to open a ticket on the client side, attaching files when opening and updating tickets on the staff side is normal.Need some help to fix this.

Tnx

ntozier commented:
@kavehsafavi What does your response have to do with this pull request?  Please open an issue report or a thread on the Official Forums at: http://osticket.com/forum

NBorba commented:
@ntozier As far as I'm concerned, the file upload problem doesn't happen with the original OsTicket. Only after we've added this feature to our install that the problem started happening. Thanks!

balojs commented:
@rayfoss please check this Feature

guilombardi91 commented:
@NBorba @kavehsafavi same issue

guilombardi91 commented:
@NBorba @kellishaver @kest874 

I found the error, it's in the javascript file, it's the last line, it has a comma +, I'm attaching the corrected file! Just copy the code

[jquery.min.txt](https://github.com/osTicket/osTicket/files/871071/jquery.min.txt)

kest874 committed:
Fix for file upload

kest874 commented:
I've committed the change. thanks @guilombardi91

naveen1410 commented:
![image](https://cloud.githubusercontent.com/assets/26955363/24752912/9bbf4726-1aed-11e7-84fb-086310279058.png)

@kest874 the help topics are not expanding and the drop down is also transparent.

Your help will be appreciated.

Server Information
osTicket Version	v1.10 (901e5ea) —  Up to date
Web Server Software	Apache/2.4.23 (Win64) PHP/5.6.25
MySQL Version	5.7.14
PHP Version	5.6.25

naveen1410 commented:
and I just noticed carefully only Top level topics are displayed :(

ambroiler commented:
Hi @kest874 , the repo is from  osticket 1.8? A clone from this last commit [95c2c8b
](https://github.com/osTicket/osTicket/pull/3250/commits/95c2c8b78a02cfe7094713151d7dcd7affb34e13) is not from 1.10, it is right?

ambroiler commented:
@naveen1410 Just correct url paths from css/helptopic.css to match your install URL.

naveen1410 commented:
@ambroiler will try that and update.

pradeep1987oct22 commented:
@khaza123  Can u please send helptopic.php file via attachement.

Thanks

gokudb commented:
I have exactly the same issue as **pradeep1987oct22** (no images or + and - in structure tree).
I'm also using version 1.10.
Is there a fix to this?
I've checked and the png file location is correctly typed in the css file.

![error](https://user-images.githubusercontent.com/30258637/28317586-e363c0e0-6bc7-11e7-9b7a-fc04edefb119.png)

gokudb commented:
I was able to fix it.
Not sure about the lower versions but in 1.10 "images" folder is in the main directory.
You need to replace all "/images" with "../images" in "helptopic.css".

The problem I have now is that it does not show any image at the very top of the tree:
![error2](https://user-images.githubusercontent.com/30258637/28318914-b2ff4b5e-6bcc-11e7-9d14-35f8c4351025.png)

Where's there a "?" sign there should be a long pauze "-".
I've noticed that there's no "blank.gif" in the original directory nor is it provided by tha autkor.
I've made one myself, however it's still the same.

It seems that "blank.gif" refers to a different image.
Any idea what image I am missing?

gokudb commented:
I've sort of figured it out.
For some reason osTicket does not read the long pause symbol int the line:
combotree('setText', '— <?php echo __('Select Help Topic'); ?> —')
in files: "ticket-open.inc.php" and "open.inc.php" (there's also one in "class.forms.php", however I'm not sure what's it for).
This happens only when using files from this mod, maybe the coding is different than the one from my translation?
A quick fix is to change the long pauses in those files to minuses (semi-long pause also did not work).

pradeep1987oct22 commented:
Thank u so much @gokudb .I had resolved it

FadhelJ commented:
Thank you so much! @kest874 @gokudb

CRDavidK commented:
In our environment, we have parent topics (such as Network) that are set to Private so that they can't be chosen as a help topic, but the subtopics are public, allowing us to group by the hidden parent (Network) and have a specific issue (Wireless, Network Speed, Resource Access, etc...) for the ticket.  When using the Tree view, only the topics which have public parents are visible.  Is there a way that private topics with public subtopics can be visible but not selectable?

mnavarrob commented:
implemented an work great in v1.10.1 (9ae093d)

just a 2 errors i can see at the moment:
1- when i o in agent panel to create new ticket the web dont load and i must do a "F5" to reload.
2- the same error " gokudb commented on 18 Jul", the "?" image, but just a visual efect i think.

thx for the great work to all ;D

Cooper513 commented:
This looks like what I've been trying to figure out but I cant seem to figure out how to get this code/mod?  I thought I cloned the right stuff but it only installs OSTicket base.  Help please?

Bob

Cooper513 commented:
Ive had some luck manually editing the files mentioned above.  I am now having the same issue as naveen1410 had above with subtopics not displaying and the background being transparent.  I suspect my issue is the missing images but can anyone confirm?  I made ghe additional changes to the long paue character and added the .. in front of /images.  I am running1.10.1.

aadrina commented:
Hi, i'm having the same issue as @Cooper513 and @naveen1410 after upgrading to newer version. Can anyone help?

enggal17 commented:
Hi all,

How to change help topic when the user was wrong to made a ticket
![Capture](https://user-images.githubusercontent.com/52996025/85821671-59101580-b7a3-11ea-9e15-0c83eda5a8c5.JPG)

ntozier commented:
Edit the ticket.
![2020-06-26_14-05-39](https://user-images.githubusercontent.com/1315768/85887793-6aaee880-b7b6-11ea-91f4-9df178a33eea.png)

Mogio99 commented:
this mod permit to have 3 level of help topic instead of two? 
Like: IT-> PC Problem -> Error Blue screen ?

Mogio99 commented:
that's mod work with 1.8 version?
with what version it work?

funkfumaster commented:
Hello ,

Can anyone confirm whether this mod works properly with v1.15x or not ?

effgarces commented:
Has anyone implemented this feature in a newer version of osticket?
